src/plugins/intel_cpu/src/nodes/memory.hpp

```diff
@@ -204,7 +219,15 @@ class MemoryInput : public MemoryInputBase {
 private:
     void assignStateHook() override {/*pass*/}
 
+    bool haveSubgraph() const {
+        return body != nullptr;
+    }
+
 private:
+    std::shared_ptr<ov::Model> body = nullptr;
+    std::unique_ptr<ov::intel_cpu::Graph> subGraph = make_unique<ov::intel_cpu::Graph>();
```

Comment:
Please don't instance `ov::intel_cpu::Graph` unconditionally. We need this to be instanced only when there is initialized `body`. In all the other cases this is simply a waste of memory.

Reply:
Done, only instance inside MemoryInput constructor when haveSubgraph.